provider/core_dhcp_options_resource.go

```diff
@@ -139,6 +147,14 @@ func (s *DHCPOptionsResourceCrud) State() string {
 }
 
 func (s *DHCPOptionsResourceCrud) Create() (e error) {
+	// If we are creating a default resource, then don't have to
+	// actually create it. Just set the ID and update it.
+	if defaultId := s.D.Get("manage_default_resource_id").(string); defaultId != "" {
```

Comment:
nit, but normally GetOk should be used to validate whether a property is set. The result should really be the same, but doesn't have to depend on the default value.

if vnicID, ok := s.D.GetOk("vnic_id"); ok {
		opts.VnicID = vnicID.(string)
	}

Reply:
I will address this. Should be easy to fix.